src/Symfony/Component/AssetMapper/ImportMap/JavaScriptImport.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\MappedAsset;
+
+/**
+ * Represents a module that was imported by a JavaScript file.
+ */
+final class JavaScriptImport
+{
+    /**
+     * @param string $importName The name of the import needed in the importmap, e.g. "/foo.js" or "react".
+     * @param bool   $isLazy     whether this import was lazy or eager
+     */
+    public function __construct(
+        public string $importName,
+        public readonly bool $isLazy = false,
+        public readonly ?MappedAsset $asset = null,
+    ) {
+    }
+
+    public function isRelativeImport(): bool
+    {
+        return str_starts_with($this->importName, '.') || str_starts_with($this->importName, '/');
```

Comment:
what about protocol-relative URLs? `//foo/bar`

Reply:
This method name was super unclear. What I wanted to say was "does this import represent a module that we should automatically add to the importmap". Changes in: https://github.com/weaverryan/symfony/commit/2a968ef97b6fca3dea28f1f309b289383ccfb2b3